### ISSUE TITLE
Code blocks may render as invalid Nix expression

### ISSUE THREAD

**Issue by hmenke** [NixOS/nixos-search#175]
The default value for the [`nix.nixPath`](https://github.com/NixOS/nixpkgs-channels/blob/c59ea8b8a0e7f927e7291c14ea6cd1bd3a16ff38/nixos/modules/services/misc/nix-daemon.nix#L413-L418) option is
```nix
let # fix the syntax highlighting by making this a valid Nix expression
        default =
          [
            "nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixos"
            "nixos-config=/etc/nixos/configuration.nix"
            "/nix/var/nix/profiles/per-user/root/channels"
          ];
```
However, this is rendered as

![test](https://user-images.githubusercontent.com/1903556/91773678-5a463e80-ec3b-11ea-984d-cc6765ca3f0b.png)

The commas between the elements are not valid Nix.

Link to the search: https://search.nixos.org/options?show=nix.nixPath&query=nixPath&from=0&size=30&sort=relevance&channel=unstable#disabled

**Comment by golddranks** [comment]
The same bug affects the `security.pam.loginLimits` example value: https://search.nixos.org/options?show=security.pam.loginLimits&query=limits&from=0&size=30&sort=relevance&channel=unstable#disabled

**Comment by mweinelt** [comment]
Another example is `services.limesurvey.virtualHost`.

```
      example = literalExample ''
        {
          hostName = "survey.example.org";
          adminAddr = "webmaster@example.org";
          forceSSL = true;
          enableACME = true;
        }
      '';
```

Rendered as:

`"{\n hostName = \"survey.example.org\";\n adminAddr = \"webmaster@example.org\";\n forceSSL = true;\n enableACME = true;\n}\n"`


https://search.nixos.org/options?show=services.limesurvey.virtualHost&query=limesurvey&from=0&size=30&sort=relevance&channel=unstable#disabled

**Comment by hmenke** [comment]
@mweinelt I've reported that previously in #161, although I think you are right and this is just another symptom of the same issue.

**Comment by garbas** [comment]
For example the example for system.activationScripts is formatted incorrectly and quite difficult to read.

**Comment by davidak** [comment]
`example = { EDITOR = "nvim"; VISUAL = "nvim"; };`

rendered as

```
Example value
{"EDITOR": "nvim", "VISUAL": "nvim"}
```

https://search.nixos.org/options?show=environment.variables&query=variables&from=0&size=30&sort=relevance&channel=20.09#disabled
https://github.com/NixOS/nixpkgs/blob/a06529b7adbfe39e06c10d0c539418d130f2fecc/nixos/modules/config/shells-environment.nix#L37

**Comment by Moredread** [comment]
`networking.interfaces` is another example:

```
      example =
        { eth0.ipv4.addresses = [ {
            address = "131.211.84.78";
            prefixLength = 25;
          } ];
        };
```

is rendered as

```
{"eth0": {"ipv4": {"addresses": [{"address": "131.211.84.78", "prefixLength": 25}]}}}
```

**Comment by ncfavier** [comment]
This was fixed in #239.

**Comment by correabuscar** [comment]
It's currently rendered as:
```
    nix.nixPath

Name nix.nixPath

Description The default Nix expression search path, used by the Nix evaluator to look up paths enclosed in angle brackets (e.g. <nixpkgs>).
Type list of string

Default ''
  if nix.channel.enable
  then [
    "nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixos"
    "nixos-config=/etc/nixos/configuration.nix"
    "/nix/var/nix/profiles/per-user/root/channels"
  ]
  else [];
''

Declared in
nixos/modules/config/nix-channel.nix
```

That is: the `Default` seems to be a multi-line string, even though the `Type` says it's not a string but a list of strings. Took me a while, as a newbie, to know to just remove the `''`s

<details>

```nix
error:
       … while calling the 'head' builtin

         at /etc/nixos/nixpkgs/lib/attrsets.nix:967:11:

          966|         || pred here (elemAt values 1) (head values) then
          967|           head values
             |           ^
          968|         else

       … while evaluating the attribute 'value'

         at /etc/nixos/nixpkgs/lib/modules.nix:809:9:

          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: A definition for option `nix.nixPath' is not of type `list of string'. Definition values:
       - In `/etc/nixos/configuration.nix':
           ''
             if config.nix.channel.enable
               then [
                 #"nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixos"
                 "nixpkgs=/etc/nixos/nixpkgs"
           ...

```

</details>

In other places it seems to show properly when it's `list of string`: https://search.nixos.org/options?channel=unstable&show=environment.profiles&from=0&size=50&sort=relevance&type=packages&query=environ
